gcc/rust/parse/rust-parse-impl.h

```diff
@@ -2078,6 +2078,79 @@ Parser<ManagedTokenSource>::parse_visibility ()
     }
 }
 
+static std::string
+extract_module_path (const AST::AttrVec &inner_attrs,
+		     const AST::AttrVec &outer_attrs, const std::string &name)
+{
```

Comment:
The current implementation of this function is in `gcc/rust/ast/rust-ast-full-test.cc`. Any reason for copying it here but no removing/renaming it from the other source?

Reply:
This one deals with inner attributes as well. The other one only deals with out-of-line modules, so there's no chance of an inner attribute affecting its path (since we're already reading the file).

Reply:
Also, the return value is different. This is "the name to use for the directory to look up submodules in". The other returns (effectively) `Option<String>` for the path to use for the module itself (with an empty string being `None`).

Reply:
Mmh. I understand the idea but there is a lot of code present in both functions. In fact, I can't really review what changed in this function versus the previously written one. Do you think you could factor them together? I'm not a fan of how a lot of the logic is repeated. We also have issues regarding cleaning up attribute handling such as https://github.com/Rust-GCC/gccrs/issues/665 so I think it's fine if factoring them together isn't perfect as it will need cleaning up anyway